build.sbt

```diff
@@ -413,14 +413,27 @@ lazy val ratSettings = Seq(
 )
 
 lazy val unidocSettings = Seq(
-  ScalaUnidoc / unidoc / unidocProjectFilter := inProjects(sapi, udf),
+  ScalaUnidoc / unidoc / unidocProjectFilter := inProjects(sapi, udf, runtime1),
   ScalaUnidoc / unidoc / scalacOptions := Seq(
     "-doc-title",
     "Apache Daffodil " + version.value + " Scala API",
     "-doc-root-content",
     (sapi / baseDirectory).value + "/root-doc.txt"
   ),
-  JavaUnidoc / unidoc / unidocProjectFilter := inProjects(japi, udf),
+  ScalaUnidoc / unidoc / unidocAllSources :=
+    (ScalaUnidoc / unidoc / unidocAllSources).value.map { sources =>
+      sources.filter { source =>
+        //
+        // This filter limits the scaladoc to only files which have
+        // api, sapi, or japi as directory names in their paths.
+        //
+        val str = source.toString
```

Comment:
`str` contains an absolute path to the sources, so there's a rare edge case that a directory that the daffodil repo is in contains one of these strings, which would result in not filtering anything out. For example, a path like `/home/user/api/repos/daffodil.git/`. I can't think of a reason why that would happen, but who knows.

We could avoid this by doing something like:
```scala
val str = source.toString.stripPrefix(baseDirectory.value.toString)
```

Reply:
Another option might be to use full paths, e.g.
```
str.conains("/org/apache/daffodil/udf/") ||
...
```
This might be better so we are more explicitly about exactly which package namespaces are considered public API.

Reply:
I will include this when I centralize the filtering.